packages/client/components/TeamPrompt/TeamPromptOptionsMenu.tsx

```diff
@@ -0,0 +1,96 @@
+import styled from '@emotion/styled'
+import graphql from 'babel-plugin-relay/macro'
+import React from 'react'
+import {useFragment} from 'react-relay'
+import useAtmosphere from '~/hooks/useAtmosphere'
+import {MenuPosition} from '~/hooks/useCoords'
+import useMenu from '~/hooks/useMenu'
+import useMutationProps from '~/hooks/useMutationProps'
+import EndTeamPromptMutation from '~/mutations/EndTeamPromptMutation'
+import {ICON_SIZE} from '~/styles/typographyV2'
+import {TeamPromptOptionsMenu_meeting$key} from '~/__generated__/TeamPromptOptionsMenu_meeting.graphql'
+import {PALETTE} from '../../styles/paletteV3'
+import CardButton from '../CardButton'
+import Icon from '../Icon'
+import IconLabel from '../IconLabel'
+import Menu from '../Menu'
+import MenuItem from '../MenuItem'
+import {MenuItemLabelStyle} from '../MenuItemLabel'
+
+const Options = styled(CardButton)({
+  position: 'absolute',
+  top: 0,
+  right: 0,
+  color: PALETTE.SLATE_700,
+  height: 32,
+  width: 32,
+  opacity: 1,
+  ':hover': {
+    backgroundColor: PALETTE.SLATE_300
+  }
+})
+
+const StyledIcon = styled(Icon)({
+  color: PALETTE.SLATE_600,
+  fontSize: ICON_SIZE.MD24,
+  marginRight: 8
+})
+
+const OptionMenuItem = styled('div')({
+  ...MenuItemLabelStyle,
+  width: '200px'
+})
+
+interface Props {
+  meetingRef: TeamPromptOptionsMenu_meeting$key
+}
+
+const TeamPromptOptionsMenu = (props: Props) => {
+  const {togglePortal, originRef, menuPortal, menuProps} = useMenu(MenuPosition.UPPER_RIGHT)
+
+  const {meetingRef} = props
+
+  const meeting = useFragment(
+    graphql`
+      fragment TeamPromptOptionsMenu_meeting on TeamPromptMeeting {
+        id
+        endedAt
+      }
+    `,
+    meetingRef
+  )
+
+  const {id: meetingId, endedAt} = meeting
+  const atmosphere = useAtmosphere()
+  const {onCompleted, onError} = useMutationProps()
+
+  const renderedMenu = (
```

Comment:
+1 it's nice to break menus out into their own components early because they almost inevitably outgrow their parents eventually. By convention I'd usually call the parent component a `Toggle` and then the one inside the portal a `Menu`

Reply:
Could you elaborate a bit on what you'd expect this to look like in the context of the top bar where this is rendered?  E.g. should these toggle + menu components be controlled from the `TopBar` where `TeamPromptOptionsMenu` is currently being rendered as an uncontrolled component, should the toggle + menu still be in their own component but just with the menu pulled into its own component, etc.

Reply:
yep, pulling the menu into its own component should suffice.  you already know the react fragment trick for handling menus, so no worries there.

A good example is how we do the dashboard topbar today (this component isn't named Toggle, but the same pattern applies): https://github.com/ParabolInc/parabol/blob/a88782807ca3a4bb1c291ec455b554f137e02476/packages/client/components/TopBarHelp.tsx

no need for the lazy preloads for small menus, but if the menus get large it's always nice to break it off into its own chunk, that way webpack can keep it out of the primary bundle & the page will load a little faster for first time visitors or when folks upgrade

Reply:
Gotcha, I think I refactored in the way you're suggesting but lmk if I misunderstood!